source/fe/mapping_isoparametric.cc

```diff
@@ -0,0 +1,2238 @@
+          &                                               cell,
+        const unsigned int                                face_no,
+        const unsigned int                                subface_no,
+        const typename QProjector<dim>::DataSetDescriptor data_set,
+        const Quadrature<dim - 1> &                       quadrature,
+        const typename dealii::MappingIsoparametric<dim, spacedim>::InternalData
+          &data,
+        internal::FEValuesImplementation::MappingRelatedData<dim, spacedim>
+          &output_data)
+      {
+        maybe_compute_q_points<dim, spacedim>(data_set,
+                                              data,
+                                              output_data.quadrature_points);
+        maybe_update_Jacobians<dim, spacedim>(CellSimilarity::none,
+                                              data_set,
+                                              data);
+        maybe_update_jacobian_grads<dim, spacedim>(CellSimilarity::none,
+                                                   data_set,
+                                                   data,
+                                                   output_data.jacobian_grads);
+        maybe_update_jacobian_pushed_forward_grads<dim, spacedim>(
+          CellSimilarity::none,
+          data_set,
+          data,
+          output_data.jacobian_pushed_forward_grads);
+        maybe_update_jacobian_2nd_derivatives<dim, spacedim>(
+          CellSimilarity::none,
+          data_set,
+          data,
+          output_data.jacobian_2nd_derivatives);
+        maybe_update_jacobian_pushed_forward_2nd_derivatives<dim, spacedim>(
+          CellSimilarity::none,
+          data_set,
+          data,
+          output_data.jacobian_pushed_forward_2nd_derivatives);
+        maybe_update_jacobian_3rd_derivatives<dim, spacedim>(
+          CellSimilarity::none,
+          data_set,
+          data,
+          output_data.jacobian_3rd_derivatives);
+        maybe_update_jacobian_pushed_forward_3rd_derivatives<dim, spacedim>(
+          CellSimilarity::none,
+          data_set,
+          data,
+          output_data.jacobian_pushed_forward_3rd_derivatives);
+
+        // TODO: pre-compute and store
+        std::vector<double> weights(quadrature.size());
+
+        const auto quad =
+          QProjector<dim>::project_to_all_faces(data.fe.reference_cell_type(),
+                                                quadrature);
+
+        for (unsigned int i = 0; i < quadrature.size(); ++i)
+          weights[i] = quad.get_weights()[i + data_set];
+
+        maybe_compute_face_data(mapping,
+                                cell,
+                                face_no,
+                                subface_no,
+                                quadrature.size(),
+                                weights,
+                                data,
+                                output_data);
```

Comment:
It is not possible to use the quadrature weights of the face quadrature rules. Instead, the projected values have to used here. These values could be pre-computed and stored in `dealii::MappingIsoparametric<dim, spacedim>::InternalData`.